modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/UuidStreamerApplication.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.ducktest.tests;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.internal.ducktest.utils.IgniteAwareApplication;
+
+/**
+ * Loading random uuids in cache.
+ */
+public class UuidStreamerApplication extends IgniteAwareApplication {
+    /** {@inheritDoc} */
+    @Override public void run(JsonNode jNode) {
+        String cacheName = jNode.get("cacheName").asText();
+
+        int dataSize = Optional.ofNullable(jNode.get("dataSize"))
+                .map(JsonNode::asInt)
+                .orElse(1024);
+
+        long iterSize = Optional.ofNullable(jNode.get("iterSize"))
+                .map(JsonNode::asLong)
+                .orElse(-1L);
+
+        CacheConfiguration<UUID, byte[]> cacheCfg = new CacheConfiguration<>(cacheName);
+        cacheCfg.setCacheMode(CacheMode.PARTITIONED);
+        cacheCfg.setBackups(2);
+        cacheCfg.setAtomicityMode(CacheAtomicityMode.TRANSACTIONAL);
+        cacheCfg.setIndexedTypes(UUID.class, byte[].class);
+
+        ignite.getOrCreateCache(cacheCfg);
+
+        long start = System.currentTimeMillis();
+
+        markInitialized();
+
+        workParallel(ignite, cacheName, iterSize, dataSize);
+
+        recordResult("DURATION", System.currentTimeMillis() - start);
+
+        markFinished();
+    }
+
+    /** */
+    private void workParallel(Ignite ignite, String cacheName, long iterSize, int dataSize) {
+        int core = Runtime.getRuntime().availableProcessors() / 2;
+
+        long iterCore = 0 < iterSize ? iterSize / core : iterSize;
+
+        ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat("UuidDataStreamer-%d")
+                .setDaemon(true)
+                .build();
+
+        ExecutorService executors = Executors.newFixedThreadPool(core, threadFactory);
+        CountDownLatch latch = new CountDownLatch(core);
+
+        for (int i = 0; i < core; i++)
+            executors.submit(new UuidDataStreamer(ignite, cacheName, latch, iterCore, dataSize));
```

Comment:
any reason to use executor service instead of threads?

Reply:
fixed, use threads